projects/electron-ngrx/src/lib/window-communication.service.ts

```diff
@@ -1,59 +1,112 @@
 import {Injectable} from '@angular/core';
 import {fromEvent, Observable} from 'rxjs';
-import {filter, map, withLatestFrom} from 'rxjs/operators';
+import {filter, map, pluck, share, take, withLatestFrom} from 'rxjs/operators';
 import {NavigationEnd, Router} from '@angular/router';
+import {ipcRenderer, remote} from 'electron';
+
+interface WithSender {
```

Comment:
name it more appropriately.
Maybe name it MessageWithSender or name it Message and the interface below MessageWithReply.
If possible - you should combine both to a single interface with optionals